examples/credential-registry-storage-contract/src/lib.rs

```diff
@@ -0,0 +1,404 @@
+/// - The signature is expired.
+/// - The signature can not be validated.
+/// - A credential is already registered for the given `public_key`.
+/// - Fails to log event.
+#[receive(
+    contract = "credential-registry-storage",
+    name = "store",
+    parameter = "StoreParam",
+    error = "CustomContractError",
+    crypto_primitives,
+    enable_logger,
+    mutable
+)]
+fn store<S: HasStateApi>(
+    ctx: &impl HasReceiveContext,
+    host: &mut impl HasHost<State<S>, StateApiType = S>,
+    logger: &mut impl HasLogger,
+    crypto_primitives: &impl HasCryptoPrimitives,
+) -> ContractResult<()> {
+    // Parse the parameter.
+    let mut cursor = ctx.parameter_cursor();
+    // The input parameter is `StoreParam` but we only read the initial part of it
+    // with `PartialStoreParam`. I.e. we read the `signature` and the
+    // `public_key`, but not the `message` here.
+    let param: PartialStoreParam = cursor.get()?;
+
+    // We read in the `message` now. `(cursor.size() - cursor.cursor_position()` is
+    // the length of the message in bytes.
+    let mut message_bytes = vec![0; (cursor.size() - cursor.cursor_position()) as usize];
+
+    cursor.read_exact(&mut message_bytes)?;
+
+    // The plan is to develop the signature schema specific to verifiable
+    // credentials in the future but for the time being, we use the signature schema
+    // that is already implemented in the Concordium browser wallet. The message
+    // signed in the Concordium browser wallet is prepended with the `account`
+    // address and 8 zero bytes. Accounts in the Concordium browser wallet can
+    // either sign a regular transaction (in that case the prepend is `account`
+    // address and the nonce of the account which is by design >= 1) or sign a
+    // message (in that case the prepend is `account` address and 8 zero bytes).
+    // Hence, the 8 zero bytes ensure that the user does not accidentally sign a
+    // transaction. The account nonce is of type u64 (8 bytes).
+    let mut msg_prepend = vec![0; 32 + 8];
+    // Prepend the `account` address of the signer.
+    msg_prepend[0..32].copy_from_slice(&param.public_key.0);
+    // Prepend 8 zero bytes.
+    msg_prepend[32..40].copy_from_slice(&[0u8; 8]);
+    // Calculate the message hash.
+    let message_hash =
+        crypto_primitives.hash_sha2_256(&[&msg_prepend[0..40], &message_bytes].concat()).0;
+
+    let message: SignedMessage = Cursor::new(&message_bytes).get()?;
+
+    // Check that the signature was intended for this contract.
+    ensure_eq!(message.contract_address, ctx.self_address(), CustomContractError::WrongContract);
+
+    // Check signature is not expired.
+    ensure!(message.timestamp > ctx.metadata().slot_time(), CustomContractError::Expired);
+
+    // Check signature was intended for this `entry_point`.
+    ensure_eq!(
+        message.entry_point.as_entrypoint_name(),
+        EntrypointName::new_unchecked("store"),
```

Comment:
I think you should be able to use just `"store"` here. I think I added the ability to check equality of entrypoint name and string.